dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/helpers.ts

```diff
@@ -0,0 +1,167 @@
+import groupBy from 'lodash/groupBy';
+import orderBy from 'lodash/orderBy';
+import { eachIsoMonthOfInterval } from '../../../../utils/date';
+import { Bank } from '../../../../types/db-models/banks';
+import { IsoMonthStamp } from '../../../../types/date';
+import { UtilisationReportReconciliationSummary, UtilisationReportReconciliationSummaryItem } from '../../../../types/utilisation-reports';
+import { UtilisationReport } from '../../../../types/db-models/utilisation-reports';
+import { UTILISATION_REPORT_RECONCILIATION_STATUS } from '../../../../constants';
+import { getAllBanks } from '../../../../services/repositories/banks-repo';
+import { getAllUtilisationDataForReport } from '../../../../services/repositories/utilisation-data-repo';
+import {
+  getUtilisationReportDetailsByBankIdMonthAndYear,
+  getOpenReportsBeforeReportPeriodForBankId,
+} from '../../../../services/repositories/utilisation-reports-repo';
+import {
+  getPreviousReportPeriodStart,
+  getReportPeriodStartForSubmissionMonth,
+  getReportPeriodStartForUtilisationReport,
+  getSubmissionMonthForReportPeriodStart,
+  isEqualReportPeriodStart,
+} from '../../../../utils/report-period';
+
+type UtilisationReportForSubmissionMonth = {
+  submissionMonth: IsoMonthStamp;
+  report: UtilisationReport | null;
+};
+
+type SummaryItemForSubmissionMonth = {
+  submissionMonth: IsoMonthStamp;
+  item: UtilisationReportReconciliationSummaryItem;
+};
+
+const mapToSummaryItem = async (bank: Bank, report: UtilisationReport | null): Promise<UtilisationReportReconciliationSummaryItem> => {
+  // TODO FN-???? - Use actual REPORT_NOT_RECEIVED UtilisationReport when added by job. Throw errors when null.
```

Comment:
this has a ticket that francesca has made now that we can probably reference just to be complete (as this will be merged into main)

Reply:
Updated to ref [FN-1949](https://ukef-dtfs.atlassian.net/browse/FN-1949)